### PR TITLE
Add player list to MP server browser.

### LINKED ISSUE #11646: In progress for -1 minute
Release-20160508.
It happens when a game has just started.

t minus one

### LINKED ISSUE #13768: Multiplayer browser: show players' names connected to a game
This way anyone will be able to spot who joined the game without joining himself. Probably should not show on closed games.
Can't say for now where to show the list though. The most simple way it seems to place a tooltip over playercount under the map pic.

### LINKED ISSUE #14569: MP filters panel is truncated at 768px vertical resolutions
I will reduce the height of the MP browser as part of my MP player list PR.

### PR DESCRIPTION
**Pull request by pchote**
This switches the game client over to the new master server protocol (https://github.com/OpenRA/OpenRAMasterServer/pull/41, https://github.com/OpenRA/OpenRAMasterServer/pull/42, https://github.com/OpenRA/OpenRAMasterServer/pull/43, https://github.com/OpenRA/OpenRAMasterServer/pull/44) and uses it to drive a new server player list.

screen shot 2017-12-24 at 15 23 24

The list is hidden for servers that do not yet use the new protocol.  Factions are hidden for servers on other mods, since we don't have the artwork to display them.

This also unifies the way that game advertisements are handled between LAN and Online games.

Closes #13768
Closes #11646
Closes #14569

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs [hunk 1]
```diff
@@ -317,7 +331,11 @@ void RefreshServerList()
 						if (addressNode != null)
 							addressNode.Value.Value = bl.Address.ToString().Split(':')[0] + ":" + addressNode.Value.Value.Split(':')[1];
 
-						lanGames.Add(new GameServer(game));
+						try
+						{
+							lanGames.Add(new GameServer(game));
+						}
+						catch { }
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs]: What is going on here? Why do we suddenly `try` `catch` this and discard any error?

File: OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs [hunk 2]
```diff
@@ -356,6 +375,85 @@ void SelectServer(GameServer server)
...
+						flag.IsVisible = () => true;
+						flag.GetImageCollection = () => "flags";
+						flag.GetImageName = () => (factionInfo != null && factionInfo.Any(f => f.InternalName == o.Faction)) ? o.Faction : "Random";
+					}
+					else
+					{
+						var label = item.Get<LabelWidget>("NOFLAG_LABEL");
+						var font = Game.Renderer.Fonts[label.Font];
+						var name = WidgetUtils.TruncateText(o.Name, label.Bounds.Width, font);
+						label.GetText = () => name;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs]: Any reason to not color the player name here as well?

**Review by abcdefg30 — DISMISSED** [review]
Looks good to me otherwise.

File: OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs (hunk 1 above)
Comment (pchote) [reply OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs]: This fixes a crash/exploit where sending malformed game advertisements will kill any other clients on the LAN.  I split this fix into its own commit to make it clearer.

File: OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs (hunk 2 above)
Comment (pchote) [reply OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs]: Done.

**Review by reaperrr — APPROVED** [review]
Code is mostly out of my league, but result looks good and seems to work fine

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e790a47a64a244ff71afa2eca681f817ccb60f3c...285c0844954f827c317c4e61f281908d765e33cc

**Pull request merged**